jmeter-service/main.go

```diff
@@ -232,6 +231,18 @@ func runWorkload(serviceURL *url.URL, testInfo *TestInfo, workload *Workload, lo
 	return executeJMeter(testInfo, workload, resultDirectory, serviceURL, resultDirectory, breakOnFunctionalIssues, logger)
 }
 
+func checkEndpointAvailable(timeout time.Duration, serviceURL *url.URL) error {
+	if serviceURL == nil {
+		return fmt.Errorf("url to check for reachability is nil")
+	}
+
+	client := http.Client{
+		Timeout: timeout,
+	}
+	_, err := client.Get(serviceURL.String())
```

Comment:
It is not guaranteed that `GET serviceURL` (e.g, `GET HTTP://carts.sockshop-dev:80/`) should work. There are plenty of services that don't return a positive HTTP result when accessing `/`.